### PR TITLE
:broom: update golang to 1.21

### PR DESCRIPTION
**Pull request by rhmdnd**

### CODE REVIEW

**Comment by BhargaviGudi** [comment]
Verification passed with 4.15.0-rc.5 + code from PR #491 

1. Install CO with PR #491 
2. Create ssb 
```
$ oc compliance bind -N test -S default-auto-apply profile/ocp4-cis profile/ocp4-cis-node
Creating ScanSettingBinding test
$ oc get suite -w
NAME   PHASE         RESULT
test   AGGREGATING   NOT-AVAILABLE
test   AGGREGATING   NOT-AVAILABLE
test   DONE          NON-COMPLIANT
test   DONE          NON-COMPLIANT
$ oc get pods
NAME                                             READY   STATUS    RESTARTS        AGE
compliance-operator-754cb8f9bf-stlh5             1/1     Running   1 (6m42s ago)   6m47s
ocp4-openshift-compliance-pp-64c669c697-msc8d    1/1     Running   0               6m39s
rhcos4-openshift-compliance-pp-97796b947-5cd8q   1/1     Running   0               6m39s
```

**Comment by rhmdnd** [comment]
This is failing serial and parallel e2e testing due to images with the wrong glibc version:

```
compliance-operator: /lib64/libc.so.6: version `GLIBC_2.32' not found (required by compliance-operator)
compliance-operator: /lib64/libc.so.6: version `GLIBC_2.34' not found (required by compliance-operator)
```

**Comment by rhmdnd** [comment]
We've been bitten by this before - here is an example of what we did to fix:

https://github.com/ComplianceAsCode/compliance-operator/pull/373

**Comment by rhmdnd** [comment]
Still failing on GLIB issues:

```
compliance-operator: /lib64/libc.so.6: version `GLIBC_2.32' not found (required by compliance-operator)
compliance-operator: /lib64/libc.so.6: version `GLIBC_2.34' not found (required by compliance-operator)
```

File: build/Dockerfile [hunk 1]
```diff
@@ -20,6 +20,7 @@ COPY --from=builder /go/src/github.com/openshift/compliance-operator/build/_outp
 
 COPY build/bin /usr/local/bin
 RUN  /usr/local/bin/user_setup
+RUN dnf update glibc -y
```
Comment (rhmdnd) [inline build/Dockerfile:23]: This shouldn't be necessary since we're already using ubi9 here.

**Comment by Vincent056** [comment]
```
compliance-operator: /lib64/libc.so.6: version `GLIBC_2.32' not found (required by compliance-operator)
compliance-operator: /lib64/libc.so.6: version `GLIBC_2.34' not found (required by compliance-operator)
```
is it because the builder image has different glibc version from the operator base image

**Comment by rhmdnd** [comment]
Checking the versions directly.

GLIBC version we were using the golang 1.20:
```
 $ podman run --entrypoint /bin/bash -ti registry.ci.openshift.org/openshift/release:rhel-8-release-golang-1.20-openshift-4.14 
[root@579d0b336b3b origin]# ldd --version
ldd (GNU libc) 2.28
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Written by Roland McGrath and Ulrich Drepper.
[root@579d0b336b3b origin]# exit
exit
```

Version with ubi8:
```
$ podman run -it registry.access.redhat.com/ubi8/ubi-micro                                                                  
sh-4.4# ldd --version
ldd (GNU libc) 2.28
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Written by Roland McGrath and Ulrich Drepper.
sh-4.4# exit
exit
```


Version we're using with golang 1.21 and RHEL 9:
```
$ podman run --entrypoint /bin/bash -ti registry.ci.openshift.org/openshift/release:rhel-9-release-golang-1.21-openshift-4.16                                                                                                                                                                                                                                                                                        [root@23c6e683bd58 origin]# ldd --version
ldd (GNU libc) 2.34
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Written by Roland McGrath and Ulrich Drepper.
[root@23c6e683bd58 origin]# exit
exit
```

Version with ubi9:
```
$ podman run -it registry.access.redhat.com/ubi9/ubi-micro                                                                   
sh-5.1# ldd --version
ldd (GNU libc) 2.34
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Written by Roland McGrath and Ulrich Drepper.
sh-5.1# exit
exit
```

**Comment by rhmdnd** [comment]
About to kick this one through https://github.com/openshift/release/pull/54429 and then we can retest this one to see if the GLIBC goes away.

**Comment by rhmdnd** [comment]
Needed to repropose the CI update in https://github.com/openshift/release/pull/54489 due to another PR reverting the update to 1.21

**Comment by rhmdnd** [comment]
Looks like we're encountering some issues with metrics that are causing the tests to fail. Need to check if this is applicable to master, or somehow related to the golang bump.

**Pull request merged**